lib/optimist.rb

```diff
@@ -97,8 +101,19 @@ def initialize(*a, &b)
     @synopsis = nil
     @usage = nil
 
+    ## allow passing settings through Parser.new as an optional hash.
+    ## but keep compatibility with non-hashy args, though.
+    begin
+      settings_hash = Hash[*a]
+      @settings = DEFAULT_SETTINGS.merge(settings_hash)
+      a=[] ## clear out args if using as settings-hash
+    rescue ArgumentError
+      @settings = DEFAULT_SETTINGS
+    end
+
     # instance_eval(&b) if b # can't take arguments
-    cloaker(&b).bind(self).call(*a) if b
+    #cloaker(&b).bind(self).call(*a) if b
```

Comment:
Should this line just be deleted?  I'm curious why you stopped using the cloaker.

Reply:
Yup, should be deleted, I just removed the call to it and the cloaker method.
AFAICT the cloaker was a super clever trick used before `instance_exec` was in ruby, since `instance_eval` didnt support passing a block, but i could be wrong about that...

In any event, instance_exec seems to work such that all tests pass and doesnt require weird trickery, so felt like the cloaker could be safely removed.